src/modules/dashboard/components/Vesting/VestingPageLayout.tsx

```diff
@@ -0,0 +1,80 @@
+import { FormikProps } from 'formik';
+import React from 'react';
+import { MessageDescriptor, defineMessage } from 'react-intl';
+import Button from '~core/Button';
+
+import Numeral from '~core/Numeral';
+import { SpinnerLoader } from '~core/Preloaders';
+import { SimpleMessageValues } from '~types/index';
+import { getFormattedTokenValue } from '~utils/tokens';
+
+import styles from './Vesting.css';
+
+const MSG = defineMessage({
+  loading: {
+    id: 'dashboard.Vesting.VestingPageLayout.loading',
+    defaultMessage: 'Loading',
+  },
+});
+
+const displayName = 'dashboard.Vesting.VestingPageLayout';
+
+interface Props {
+  title: JSX.Element;
+  buttonText: MessageDescriptor | string;
+  buttonTextValues?: SimpleMessageValues;
+  tableValues: {
+    label: JSX.Element;
+    value: string;
+    id: number;
+  }[];
+  tokenDecimals: number;
+  isLoading: boolean;
+}
+
+const VestingPageLayout = ({
+  title,
+  tableValues,
+  buttonText,
+  tokenDecimals,
+  buttonTextValues,
+  isLoading,
+  isSubmitting,
+  handleSubmit,
+}: Props & FormikProps<{}>) => {
+  return isLoading ? (
+    <div className={styles.loader}>
+      <SpinnerLoader
+        loadingText={MSG.loading}
+        appearance={{ theme: 'primary', size: 'massive' }}
+      />
+    </div>
+  ) : (
+    <div className={styles.main}>
+      {title}
+      <div className={styles.table}>
+        {tableValues.map(({ label, value }) => (
+          <div className={styles.item} key={value}>
```

Comment:
This can't work properly as a `key` since those values will at some point be equal _(ie: two or more are `0`)_

The approach to take here, is to set up an autoID generator, like we have in places where we don't actually have access to good ids.

See this example:
https://github.com/JoinColony/colonyDapp/blob/f8080fb95a6b3f5c556355870f192edbabf9b41f/src/modules/core/components/IndexModal/IndexModal.tsx#L36-L42